plugins/modules/ndo_interface_setting.py

```diff
@@ -559,7 +560,7 @@ def main():
         if uuid and not mso.existing:
             mso.fail_json(msg="{0} with the UUID: '{1}' not found".format(object_description, uuid))
 
-        if mso.existing:
+        if mso.existing and match:
```

Comment:
It should be `match` or `interface_path`?

You can remove the `match` from the check when you check `mso.existing`.

Reply:
I mentioned in a previous comment that I added the extra conditional so we can safely handle unbound variable scenarios. The parts below this conditional all use `match` and the `_path` variables. However, the `_path` variables are only set when there is a `match`. Its possible for `mso.existing` to be set without a match when name/UUID isn't provided, even though that doesn't usually happen. 

I could check all 3 variables `if mso.existing and match and interface_path`. Although, I think existing and `match` is enough here.